spec/rubocop/cop/style/redundant_array_flatten_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::RedundantArrayFlatten, :config do
+  it 'registers an offense for `x.flatten.join`' do
+    expect_offense(<<~RUBY)
+      x.flatten.join
+        ^^^^^^^^^^^^ Prefer `Array#join` without `#flatten` to join nested arrays.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      x.join
+    RUBY
+  end
+
+  it 'registers an offense for `x.flatten.join(separator)`' do
+    expect_offense(<<~RUBY)
+      x.flatten.join(separator)
+        ^^^^^^^^^^^^^^^^^^^^^^^ Prefer `Array#join` without `#flatten` to join nested arrays.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      x.join(separator)
+    RUBY
+  end
+
+  it 'registers an offense for `x.flatten(depth).join`' do
+    expect_offense(<<~RUBY)
+      x.flatten(depth).join
+        ^^^^^^^^^^^^^^^^^^^ Prefer `Array#join` without `#flatten` to join nested arrays.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      x.join
+    RUBY
+  end
+
+  it 'registers an offense for `x&.flatten&.join`' do
+    expect_offense(<<~RUBY)
+      x&.flatten&.join
+         ^^^^^^^^^^^^^ Prefer `Array#join` without `#flatten` to join nested arrays.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      x&.join
+    RUBY
+  end
+
+  it 'does not register an offense for `x.flatten.foo`' do
+    expect_no_offenses(<<~RUBY)
+      x.flatten.foo
+    RUBY
+  end
```

Comment:
Can you add the test cases?

```suggestion
  end

  it 'does not register an offense for `x.flatten(depth, extra_arg).join`' do
    expect_no_offenses(<<~RUBY)
      x.flatten(depth, extra_arg).join
    RUBY
  end

  it 'does not register an offense for `x.flatten.join(separator, extra_arg)`' do
    expect_no_offenses(<<~RUBY)
      x.flatten.join(separator, extra_arg)
    RUBY
  end
```